WalletWasabi/Models/Cluster.cs

```diff
@@ -0,0 +1,47 @@
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using WalletWasabi.Bases;
+
+namespace WalletWasabi.Models
+{
+	public class Cluster : NotifyPropertyChangedBase
+	{
+		private List<SmartCoin> Coins { get; set; }
+
+		private string _labels;
+
+		public Cluster(SmartCoin coin)
+		{
+			Labels = "";
+			Coins = new List<SmartCoin>();
+			Coins.Add(coin);
+			Labels = string.Join(", ", coin.Label.Labels);
+		}
+
+		public string Labels
+		{
+			get => _labels;
+			private set => RaiseAndSetIfChanged(ref _labels, value);
+		}
+
+		public void Merge(Cluster clusters)
+		{
+			Merge(clusters.Coins);
+		}
+
+		public void Merge(IEnumerable<SmartCoin> coins)
```

Comment:
Are there tests for this function?

Reply:
Yes, there are many. One for every possible clustering scenario. In fact, I did a change in the last minute in this code and the tests failed ;)